src/views/AddItem.jsx

```diff
@@ -1,3 +1,101 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import { useState } from 'react';
+import { addItem } from '../api/firebase';
+
+export function AddItem({ listToken }) {
+	const [daysUntilNextPurchase, setTimeFrame] = useState('7');
+	const [itemName, setItem] = useState('');
+	const [itemData, setData] = useState({});
+	const [error, setError] = useState(false);
+	const [success, setSuccess] = useState(false);
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		setData({ itemName, daysUntilNextPurchase });
+		try {
+			await addItem(listToken, { itemName, daysUntilNextPurchase });
+			setError(false);
+			setSuccess(true);
+		} catch (err) {
+			console.log(err.message);
+			setError(true);
+		}
+	};
+
+	const handleName = (e) => {
+		setSuccess(false);
+
+		setItem(e.target.value);
+	};
+
+	const handleTime = (e) => {
+		setTimeFrame(e.target.value);
+		setSuccess(false);
+	};
+	// TODO: require input for item name - done [V]
+	return (
+		<div>
+			<form onSubmit={handleSubmit}>
+				<div className="item-name">
+					<label htmlFor="item-name">
+						Item Name:{' '}
+						<input
+							required
+							value={itemName}
+							type="text"
+							name="item-name"
+							id="item-name"
+							placeholder="name of item"
+							onChange={handleName}
+						/>
+					</label>
+				</div>
+				<div>
+					<fieldset className="timeframe">
+						<legend className="legend">
+							How soon will you buy this again?
+						</legend>
+						<label htmlFor="soon">
+							<input
+								type="radio"
+								value="7"
+								checked={daysUntilNextPurchase === '7'}
+								name="time-frame"
+								id="soon"
+								onChange={handleTime}
+							/>
+							Soon
+						</label>
+
+						<label htmlFor="kind-of-soon">
+							<input
+								type="radio"
+								name="time-frame"
+								id="kind-of-soon"
+								value="14"
+								onChange={handleTime}
+							/>
+							Kind of Soon
+						</label>
+
+						<label htmlFor="not-soon">
+							<input
+								type="radio"
+								name="time-frame"
+								id="not-soon"
+								value="30"
+								onChange={handleTime}
+							/>
+							Not Soon
+						</label>
+					</fieldset>
+				</div>
+				{error && <p>The item was not added</p>}
```

Comment:
I saw your note/question regarding these messages persisting until a user adds another item (or starts typing/updating the state). Wondering if we could tuck both of these messages into a setTimeout function? I don't have a strong opinion about this, but it could be useful to show for a couple of seconds and then disappear?

Reply:
I like the success and error handling and I think it could be nice to make it more reactive, I think there are issues with setTimeout and accessibility especially making sure users have plenty of time to review notifications. I know there are also issues in React with mounting and unmounting if a timeout has that component persisting when a user navigates away. 

Nick and Zenzi can probably give more advice and feedback here and there are many experienced accessibility specialists in the collab lab community -- I would definitely advise you to start a thread in one of the public slack channels if you're interested in exploring solutions to these type of notification messages and functionality.

<details><summary>I know how to avoid setTimeout with conditional rendering example below
  </summary>

1 Setting a state hook
const [isEditMode, setIsEditMode] = useState(false)

2. changing state in both handlechange and handlesubmit
handleChange=(e) => {
... some code..
setIsEditMode(true)
}

handleSubmit=(e)=>{
..some code..
//item gets submitted to db
setIsEditMode(false)
}

3. in render 
{isEditMode && error ? <p>The item was not added</p> : <p>Please add an item</p>}

(I did an else return for no error and not in edit mode because I know conditionally rendering null as the else can also be problematic with accessibility and screen navigation, like in React with the virtual dom screen readers and keyboards can lose the element their tab is sitting on if things are conditionally rendered and there aren't proper input refs.. I'm still learning about this as well but some context.)
</details>

Reply:
@michellerenehey and @lindseyindev thanks for your suggestions! I think we could also add automatic clearing of the input box when editing starts

Reply:
I implemented this differently - the input gets cleared when the user hits the submit button and the timeframe selection returns to the default option ('soon'). The code with setSuccess and setError placed in corresponding sections already provides the transition between the editing and non-editing states, so introducing a new variable would be too much, but it could make the code easier to read - I am open to discussion